bindings/pyroot_experimental/PyROOT/src/DeclareCallablePyz.cxx

```diff
@@ -0,0 +1,663 @@
+{
+   // Parse arguments and extract input and output types and an optional verbose flag
+   if (!PyTuple_Check(args)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Invalid tuple.");
+      return NULL;
+   }
+
+   const auto size = PyTuple_Size(args);
+   if (!(size == 3 || size == 4)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Expect exactly two arguments (list of input types, return type, optional name).");
+      return NULL;
+   }
+
+   auto instance = PyTuple_GetItem(args, 0);
+   auto inputTypes = PyTuple_GetItem(args, 1);
+   auto returnType = PyTuple_GetItem(args, 2);
+
+   // Attach arguments to instance
+   PyObject_SetAttrString(instance, "input_types", inputTypes);
+   PyObject_SetAttrString(instance, "return_type", returnType);
+
+   // Attach optional name to instance
+   if (size == 4) {
+      auto name = PyTuple_GetItem(args, 3);
+      if (!CPyCppyy_PyUnicode_Check(name)) {
+         PyErr_SetString(PyExc_RuntimeError,
+                 "Failed to parse arguments: Given name is not a valid string.");
+         return NULL;
+      }
+      PyObject_SetAttrString(instance, "name", name);
+   }
+
+   Py_RETURN_NONE;
+}
+
+
+// Call method of class used as decorator to create generic C++ wrapper
+// The call method creates the C++ wrapper class for the Python callable and
+// passes through the actual callable.
+PyObject* GenericCallableImpl_call(PyObject * /*self*/, PyObject *args)
+{
+   // Parse arguments with Python callable and get attributes with types
+   if (!PyTuple_Check(args)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Invalid tuple.");
+      return NULL;
+   }
+
+   if (!(PyTuple_Size(args) == 2)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Expect exactly one argument (Python callable).");
+      return NULL;
+   }
+
+   if (!(PyTuple_Size(args) == 2)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Expect exactly one argument (Python callable).");
+      return NULL;
+   }
+
+   auto instance = PyTuple_GetItem(args, 0);
+   auto pyfunc = PyTuple_GetItem(args, 1);
+   Py_INCREF(pyfunc);
+
+   if (!PyObject_HasAttrString(instance, "input_types")) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to create C++ callable: No input_types attribute found.");
+      return NULL;
+   }
+
+   if (!PyObject_HasAttrString(instance, "return_type")) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to create C++ callable: No return_type attribute found.");
+      return NULL;
+   }
+
+   auto inputTypes = PyObject_GetAttrString(instance, "input_types");
+   auto returnType = PyObject_GetAttrString(instance, "return_type");
+
+   // Extract name of Python callable
+   PyObject* pyname;
+   if (PyObject_HasAttrString(instance, "name")) {
+      pyname = PyObject_GetAttrString(instance, "name");
+   } else {
+      if (!PyObject_HasAttrString(pyfunc, "__name__")) {
```

Comment:
Can you add a comment in this `else` case? "If the user specified no name for the C++ callable, we just use the name of the Python callable" or similar